rainier-core/src/main/scala/com/stripe/rainier/core/Injection.scala

```diff
@@ -35,37 +37,56 @@ trait Injection { self =>
 }
 
 /**
-  * Class to scale a distribution under multiplication by a scale factor
+  * Class to scale a distribution under multiplication by a positive scale factor.
+  * We assume that (a > 0).
   */
 final case class Scale(a: Real) extends Injection {
-  private val lj = a.log * -1
   def forwards(x: Real): Real = x * a
   def backwards(y: Real): Real = y / a
-  def logJacobian(y: Real): Real = lj
+  def logJacobian(y: Real): Real = a.log * -1
```

Comment:
Is there a reason you got rid of the shared object for this? I try to be careful about not over-allocating where possible.

Reply:
`lj` wasn't being used anywhere else so I thought it should be inlined... what do you mean by over-allocating here?

Reply:
Ah, putting it in a `val` will make it calculate only once, right?

Reply:
exactly